### DISCUSSION TITLE
Differences between ArchR and Signac

### DISCUSSION THREAD

**Discussion by cv55** [GreenleafLab/ArchR#1304] [Questions / Documentation]
Hello,
i just started using ArchR as a more versatile alternative of Signac, and I noticed a few differences when analysing the same dataset. First the number of cells ArchR retrieved from the fragments file (output from cellranger atac) was much higher compared to the number of cells called from cellranger.
Second, the TSS estimation was higher when using ArchR (for some cells the TSS enrichment score estimated from ArchR was 3 times higher that the TSS score from Signac). I am using the same reference hg38, following the commands shown in the respective tutorials.
Is it something you have also observed?

Thanks,
Christos

**Comment by rcorces** [comment]
Thanks for using ArchR!

> the number of cells ArchR retrieved from the fragments file (output from cellranger atac) was much higher compared to the number of cells called from cellranger.

Yes - it is expected that the number of cells identified by ArchR would be different from cellranger. cellranger estimates what a cell is in a very different way. Its still important to look at the plot of TSS enrichment x frags and make sure you use appropriate cutoffs for your data.

> Second, the TSS estimation was higher when using ArchR (for some cells the TSS enrichment score estimated from ArchR was 3 times higher that the TSS score from Signac). I am using the same reference hg38, following the commands shown in the respective tutorials.
Is it something you have also observed?

Three-times higher sounds like a lot but (at least originally) Signac uses only a small fraction of the genome to calculate the TSS enrichment. I've never really used Signac so I dont have a good comparison to offer.

**Comment by DaianeH** [comment]
Hi, I observed the same in my data, the same pattern across all 59 datasets of my project.

In one example dataset, 20k nuclei were included in the scATAC-seq experiment, but archR reads over 45k cells. Signac on the other hand reads the exact number that CellRanger says are in the dataset, which is around 10k. I imagine this is because Signac must do most of the cleaning and filtering under the hood that CellRanger does, while archR just reads everything.

I intersected the exact same cells (by barcode) from a Signac project and an archR project, and plotted the following QC plots:

![TSS_archr](https://user-images.githubusercontent.com/2428375/220680134-e284628b-265d-4692-80de-0f192abf8e4b.png)
![TSS_signac](https://user-images.githubusercontent.com/2428375/220680071-fd7d5cf8-4f8e-47e0-a018-79536d553103.png)
a4f.png)

You can see that there is indeed a big difference in values of TSS enrichment - despite the fact that these are the exact same cells, so it seems indeed that Signac and archR calculate TSS enrichment differently.

**Comment by cv55** [comment]
Coming back to this, the cell count difference is because Signac reads the filtered matrix from CellRanger whereas ArchR reads the fragments file that contains filtered and unfiltered cells from CellRanger. 
My interpretation for the difference in the TSS scores is that ArchR is using a broader region around the TSS (I think 4000 bp in total) but Signac uses only 2000bp, which explains the difference.

**Comment by nigiord** [comment]
I investigated a little, there are actually a lot of things that changes between Signac and ArchR regarding TSSEnrichment. You can find a detailed description [here](https://github.com/stuart-lab/signac/issues/1638#issuecomment-2247299507) but basically:

- As mentioned, ArchR does not use the cellranger-filtered barcodes contrary to Signac tutorial so you will have more barcodes
- ArchR uses TxDb gene annotations by default to get position of TSS, while Signac tutorial uses Ensembl annotations (so the position and number of TSS could slightly change)
- Signac uses only the TSS with `protein_coding` annotation from Ensembl, while ArchR uses all transcripts present in TxDb
- ArchR computes the TSS enrichment as follows (by default): 101 bp centered on TSS divided by 2x100 bp at the flanks of a 4000-bp-long distribution. Signac uses by default a 2000-bp-long distribution, but more importantly uses 1001 bp centered on the TSS as the signal numerator. This is HUGE (50% of the distribution) and would definitively result in smaller TSS enrichment values for all barcodes.

Currently, one way to compare both algorithms on your data would be to adapt the TSSParams of ArchR to comply with the Signac definition since Signac does not provide a way to change those parameters. But I find the ArchR definition more in line with ENCODE definition though, and I hope Signac developers will at least allow to change those parameters.
